examples/minwebgl/hexagonal_grid/src/main.rs

```diff
@@ -76,347 +75,326 @@ fn draw_hexes() -> Result< (), minwebgl::WebglError >
+      hex_shader.uniform_upload( "u_color", &[ 0.1, 0.1, 0.1, 1.0 ] );
+      hexagon_geometry.activate();
+      context.draw_arrays( GL::TRIANGLES, 0, hexagon_geometry.nvertices );
+    }
+
+    let goal = selected_hex_coord;
+
+    let path = pathfind::find_path
+    (
+      start,
+      &goal,
+      | coord | obstacles.get( &coord ).copied().unwrap_or_default()
+    );
 
-      let closure = move | input : &input::InputState, _ |
+    if let Some( ( path, _ ) ) = path
+    {
+      // draw path
+      for coord in path
       {
-        let is_mouse_down = input.is_button_down( MouseButton::Main );
-
-        if is_mouse_down
-        {
-          let rect = canvas.get_bounding_client_rect();
-          let canvas_pos = F32x2::new( rect.left() as f32, rect.top() as f32 );
-          let half_size : F32x2 = canvas_size / 2.0;
-          let cursor_pos = F32x2::new( input.mouse_position()[ 0 ] as f32, input.mouse_position()[ 1 ] as f32 );
-          let cursor_pos : Pixel = ( ( ( cursor_pos - canvas_pos ) - half_size ) / ( half_size * aspect_scale ) ).into();
-          let selected_hex_coord : Coordinate::< Axial, Pointy > = cursor_pos.into();
-
-          let color = color_picker.value();
-          let r = u8::from_str_radix( &color[ 1..3 ], 16 ).unwrap() as f32 / 255.0;
-          let g = u8::from_str_radix( &color[ 3..5 ], 16 ).unwrap() as f32 / 255.0;
-          let b = u8::from_str_radix( &color[ 5..7 ], 16 ).unwrap() as f32 / 255.0;
-          let color = [ r, g, b ];
-
-          if painting_canvas[ selected_hex_coord ] == color { return; }
-
-          painting_canvas[ selected_hex_coord ] = color;
-
-          // draw painted hexagon
-          let axial : Coordinate< Axial, _ > = selected_hex_coord.into();
-          let hex_pos : Pixel = axial.into();
-          let translation = mat2x2h::translate( [ hex_pos[ 0 ], -hex_pos[ 1 ] ] );
-          let mvp = scale_m * translation * mat2x2h::rot( 30.0f32.to_radians() );
-          hex_shader.uniform_matrix_upload( "u_mvp", mvp.raw_slice(), true );
-          hex_shader.uniform_upload( "u_color", &[ r, g, b, 1.0 ] );
-          hexagon_geometry.activate();
-          context.draw_arrays( GL::TRIANGLES, 0, hexagon_geometry.nvertices );
-        }
-      };
-
-      input.borrow_mut().add_callback
-      (
-        closure,
-        EventFlags::MouseMovement | EventFlags::MouseButton,
-      );
+        let hex_pos : Pixel = coord.into();
+        let translation = mat2x2h::translate
+        (
+          [ hex_pos[ 0 ] - grid_center[ 0 ], -hex_pos[ 1 ] + grid_center[ 1 ] ]
+        );
+        let mvp = scale_m * translation * mat2x2h::rot( 30.0f32.to_radians() ); // * mat2x2h::scale( [ 0.9, 0.9 ] );
+
+        hex_shader.uniform_matrix_upload( "u_mvp", mvp.raw_slice(), true );
+        hex_shader.uniform_upload( "u_color", &[ 0.1, 0.6, 0.1, 1.0 ] );
+        hexagon_geometry.activate();
+        context.draw_arrays( GL::TRIANGLES, 0, hexagon_geometry.nvertices );
```

Comment:
allocating new matrices in loop is expansive and slow, can we do allocation outside and do only change in loop?